app/ide-desktop/lib/client/src/index.ts

```diff
@@ -205,6 +225,41 @@ class App {
             })
         }
         electron.ipcMain.on(ipc.channel.quit, () => electron.app.quit())
+        // FIXME [NP]: Move this to the authentication package.
+        electron.ipcMain.on(ipc.channel.openExternalUrl, (event, url) => opener(url))
+        //electron.ipcMain.on(ipc.channel.setAuthenticatedRedirectCallback, (event, callback) => callback())
+    }
+
+    /**
+     * Initialize the listener for `open-url` events.
+     *
+     * This listener is used to open a page in *this* application window, when the user is
+     * redirected to a URL with a protocol supported by this application.
+     *
+     * For example, when the user completes an OAuth sign in flow (e.g., through Google), they are
+     * redirected to a URL like `enso://auth?code=...`. This listener will intercept that URL
+     * and open the page `auth?code=...` in the application window.
+     */
+    initOpenUrlListener() {
+        electron.app.on('open-url', (event, url) => {
+            // Prevent the default behavior (don't open `enso://auth?code=...` URL in the window).
```

Comment:
This comment is misleading. It prevents. not only this url from opening in browser, but other urls as well.